### PR TITLE
Added test to catch unresolved promise when job not present.

### PR DESCRIPTION
**Pull request by jwpf100**

### CODE REVIEW

File: src/daos/job.js [hunk 1]
```diff
@@ -62,7 +62,11 @@ class JobDao {
    */
   async fetchJobById(jobId) {
     const jobRow = await this.JobStore.fetchJobById(jobId);
-    return this._JobRowtoJobData(jobRow);
+    if (jobRow !== undefined) {
+      return this._JobRowtoJobData(jobRow);
+    } else {
+      return null;
+    }
```
Comment (MattGibney) [inline src/daos/job.js:69]: Please change this to use the return early approach instead.

File: src/controllers/api.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ module.exports = {
    */
   async fetchJob(req, res) {
     const jobId = parseInt(req.params.jobId);
-
+    console.log('controller sthttp-sart');
```
Comment (MattGibney) [inline src/controllers/api.js:87]: Remove console.log

File: src/models/job.js [hunk 3]
```diff
@@ -76,6 +76,7 @@ class JobModel {
    * @returns {Promise<JobModel[]|null>} Array of JobModel Objects. Null on error.
    */
   static async fetchAllJobs(ModelFactory, DaoFactory) {
+    console.log('model start');
```
Comment (MattGibney) [inline src/models/job.js:79]: remove console.log

File: test/daos/job.spec.js [hunk 4]
```diff
@@ -64,6 +64,17 @@ describe('Dao: Job', function () {
 
       assert.equal(jobData.id, 1);
     });
+    it('responds null when db returns undefined', async function () {
+      const mockJobStore = {
+        fetchJobById: sinon.stub().returns(undefined),
```
Comment (MattGibney) [inline test/daos/job.spec.js:69]: fetchJobById is asynchronous and returns a promise, so you should switch this line to:
`fetchJobById: sinon.stub().resolves(undefined),`

**Review by MattGibney — CHANGES_REQUESTED** [review]

File: test/daos/job.spec.js (hunk 4 above)
Comment (jwpf100) [reply test/daos/job.spec.js]: Done

File: src/models/job.js (hunk 3 above)
Comment (jwpf100) [reply src/models/job.js]: Done

File: src/controllers/api.js (hunk 2 above)
Comment (jwpf100) [reply src/controllers/api.js]: Done

File: src/daos/job.js (hunk 1 above)
Comment (jwpf100) [reply src/daos/job.js]:
```
async fetchJobById(jobId) {
    const jobRow = await this.JobStore.fetchJobById(jobId);
    if (jobRow === undefined) {
      return null;
    }
    return this._JobRowtoJobData(jobRow);
  }
```

**Review by MattGibney — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**